src/test/java/io/nats/client/impl/JetStreamManagementTests.java

```diff
@@ -1368,4 +1430,128 @@ public void testDirectMessageRepublishedSubject() throws Exception {
+            ConsumerConfiguration cc22 = ConsumerConfiguration.builder().name(name).filterSubjects(fs2).build();
+            ci = jsmNew.updateConsumer(stream2, cc22);
+            assertEquals(name, ci.getName());
+            assertEquals(fs2, ci.getConsumerConfiguration().getFilterSubject());
+
+            // --------------------------------------------------------
+            // Pre 290 durable pathway
+            // --------------------------------------------------------
+            String stream3 = streamPrefix + "-old-durable";
+            name = name();
+            subject = name();
+            fs1 = subject + ".A";
+            fs2 = subject + ".B";
+            String fs3 = subject + ".C";
+            createMemoryStream(jsmPre290, stream3, subject + ".*");
+
+            ConsumerConfiguration cc31 = ConsumerConfiguration.builder().durable(name).filterSubject(fs1).build();
+
+            // update no good when not exist
+            e = assertThrows(JetStreamApiException.class, () -> jsmPre290.updateConsumer(stream3, cc31));
+            assertEquals(10149, e.getApiErrorCode());
+
+            // initial create ok
+            ci = jsmPre290.createConsumer(stream3, cc31);
+            assertEquals(name, ci.getName());
+            assertEquals(fs1, ci.getConsumerConfiguration().getFilterSubject());
+
+            // opt out of 209, create on existing ok
+            // This is not exactly the same behavior as with the new consumer create api, but it's what the server does
+            jsmPre290.createConsumer(stream3, cc31);
+
+            ConsumerConfiguration cc32 = ConsumerConfiguration.builder().durable(name).filterSubject(fs2).build();
+            e = assertThrows(JetStreamApiException.class, () -> jsmPre290.createConsumer(stream3, cc32));
+            assertEquals(10148, e.getApiErrorCode());
+
+            // update ok when exists
+            ConsumerConfiguration cc33 = ConsumerConfiguration.builder().durable(name).filterSubjects(fs3).build();
+            ci = jsmPre290.updateConsumer(stream3, cc33);
+            assertEquals(name, ci.getName());
+            assertEquals(fs3, ci.getConsumerConfiguration().getFilterSubject());
+
+            // --------------------------------------------------------
+            // Pre 290 ephemeral pathway
+            // --------------------------------------------------------
+            subject = name();
+
+            String stream4 = streamPrefix + "-old-ephemeral";
+            fs1 = subject + ".A";
+            createMemoryStream(jsmPre290, stream4, subject + ".*");
+
+            ConsumerConfiguration cc4 = ConsumerConfiguration.builder().filterSubject(fs1).build();
+
+            // update no good when not exist
+            e = assertThrows(JetStreamApiException.class, () -> jsmPre290.updateConsumer(stream4, cc4));
+            assertEquals(10149, e.getApiErrorCode());
+
+            // initial create ok
+            jsmPre290.createConsumer(stream4, cc4);
```

Comment:
The line below currently fails the assert since `ci` is not updated, adding that here should make the test pass.
```suggestion
            ci = jsmPre290.createConsumer(stream4, cc4);
```